frontend/simple/views/SignUp.vue

```diff
@@ -65,35 +69,64 @@
 </style>
 
 <script>
-var serialize = require('form-serialize')
-var request = require('superagent')
+import Vue from 'vue'
+import * as Events from '../../../shared/events'
+import * as db from '../js/database'
+import multihash from 'multihashes'
+import {HapiNamespace} from '../js/backend/hapi'
+var namespace = new HapiNamespace()
 
 // TODO: fix all this
 export default {
   name: 'SignUp',
   methods: {
-    submit: function () {
-      console.log(this.errors)
-      this.response = ''
-      request.post(`${process.env.API_URL}/user/`)
-      .send(serialize(this.$refs.form, {hash: true}))
-      .end((err, res) => {
-        this.error = !!err || !res.ok
-        console.log('this.error', this.error)
-        this.response = this.error ? res.body.message : (res.text === '' ? 'success' : res.text)
-        if (!this.error && this.$route.query.next) {
-          this.login()
-          this.$router.push({path: this.$route.query.next})
+    submit: async function () {
+      try {
+        let user = new Events.IdentityContract({
+          attributes: [
+            {name: 'name', value: this.name},
+            {name: 'email', value: this.email}
+          ]
+        })
+        await namespace.register(this.name, user.toHash())
+        await db.storeLogin(user.toHash(), multihash.toB58String(multihash.encode(new Buffer(this.password), 'blake2b')))
+        this.response = 'success'
+        if (this.$route.query.next) {
+          setTimeout(() => {
+            this.$router.push({path: this.$route.query.next})
+          }, 1000)
+        }
+        this.$store.commit('login', this.name)
+        this.error = false
+      } catch (ex) {
+        console.log(ex)
+        this.response = ex.toString()
+        this.error = true
+      }
+    },
+    forwardToLogin: function () {
+      Vue.events.$emit('login')
+    },
+    checkName: async function () {
```

Comment:
Let's have this called via a debounced manner each time the field changes. I.e. no more than calling it once per 2 seconds. If you search the vuejs guide for "debounce" you'll get several results, including this one:

- https://vuejs.org/v2/guide/migration.html#debounce-Param-Attribute-for-v-model-removed

Notice also the note at the bottom regarding throttle. Decide which provides a better experience and go for it. For importing debounce or throttle, see how it's done in `state.js` for `debouncedSave`.